src/NUnitFramework/framework/Internal/Builders/DefaultTestCaseBuilder.cs

```diff
@@ -105,36 +106,45 @@ public Test BuildFrom(IMethodInfo method, Test? parentSuite)
         {
             var tests = new List<TestMethod>();
 
-            var metadata = MethodInfoCache.Get(method);
+            try
+            {
+                var metadata = MethodInfoCache.Get(method);
+
+                List<ITestBuilder> builders = new List<ITestBuilder>(metadata.TestBuilderAttributes);
 
-            List<ITestBuilder> builders = new List<ITestBuilder>(metadata.TestBuilderAttributes);
+                // See if we need to add a CombinatorialAttribute for parameterized data
+                if (method.MethodInfo.GetParameters().Any(param => param.HasAttribute<IParameterDataSource>(false))
+                    && !builders.Any(builder => builder is CombiningStrategyAttribute))
+                    builders.Add(new CombinatorialAttribute());
 
-            // See if we need to add a CombinatorialAttribute for parameterized data
-            if (method.MethodInfo.GetParameters().Any(param => param.HasAttribute<IParameterDataSource>(false))
-                && !builders.Any(builder => builder is CombiningStrategyAttribute))
-                builders.Add(new CombinatorialAttribute());
+                foreach (var attr in builders)
+                {
+                    foreach (var test in attr.BuildFrom(method, parentSuite))
+                        tests.Add(test);
+                }
 
-            foreach (var attr in builders)
+                return builders.Count > 0 && method.GetParameters().Length > 0 || tests.Count > 0
+                    ? BuildParameterizedMethodSuite(method, tests)
+                    : BuildSingleTestMethod(method, parentSuite);
+            }
+            catch (Exception ex)
             {
-                foreach (var test in attr.BuildFrom(method, parentSuite))
-                    tests.Add(test);
+                var testMethod = new TestMethod(method, parentSuite);
+                testMethod.MakeInvalid("An exception was thrown while loading the test." + Environment.NewLine + ex.ToString());
```

Comment:
nitpick/question: For both `TestMethod` and `TestFixture` in the file above, this method is defined on a base class, `Test`. `Test.MakeInvalid` currently only has one overload, which takes a string. Is it worth adding another which can take in both a string and an Exception?

```csharp
public abstract class Test {
  // Should this overload be public or internal? Not sure
  public void MakeInvalid(string reason, Exception ex) {
    // Something happens here, probably just building the message from "reason" and "ex"
  }
}
```
 It seems like we have to handle this scenario a few times within this PR

Reply:
Overload added.
Also split the StackTrace from the Message, setting separate StackTrace property.
Updated test.